crml/cennzx-spot/src/lib.rs

```diff
@@ -294,9 +280,8 @@ decl_module! {
 
 			<pallet_generic_asset::Module<T>>::make_transfer(&core_asset_id, &exchange_address, &from_account, core_asset_amount)?;
 			<pallet_generic_asset::Module<T>>::make_transfer(&asset_id, &exchange_address, &from_account, trade_asset_amount)?;
-			Self::set_liquidity(&exchange_key, &from_account,
-									account_liquidity - liquidity_withdrawn);
-			Self::burn_total_supply(&exchange_key, liquidity_withdrawn);
+			Self::set_liquidity(&exchange_key, &from_account, account_liquidity - liquidity_to_withdraw);
```

Comment:
Seems like instead of `set_liquidity` we could do `burn_liquidity` and `mint_liquidity` and within that do the accounting for total_liquidity aswell

Reply:
I like that, let me try it out